plugins/database/redshift/redshift_test.go

```diff
@@ -48,281 +45,226 @@ var (
 	vaultACC = "VAULT_ACC"
 )
 
-func redshiftEnv() (url string, user string, password string, errEmpty error) {
-	errEmpty = errors.New("err: empty but required env value")
+func interpolateConnectionURL(url, user, password string) string {
+	return fmt.Sprintf("postgres://%s:%s@%s", user, password, url)
+}
 
+func redshiftEnv() (connURL string, url string, user string, password string, errEmpty error) {
 	if url = os.Getenv(keyRedshiftURL); url == "" {
-		return "", "", "", errEmpty
+		return "", "", "", "", fmt.Errorf("%s environment variable required", keyRedshiftURL)
 	}
 
 	if user = os.Getenv(keyRedshiftUser); url == "" {
-		return "", "", "", errEmpty
+		return "", "", "", "", fmt.Errorf("%s environment variable required", keyRedshiftUser)
 	}
 
 	if password = os.Getenv(keyRedshiftPassword); url == "" {
-		return "", "", "", errEmpty
+		return "", "", "", "", fmt.Errorf("%s environment variable required", keyRedshiftPassword)
 	}
 
-	url = fmt.Sprintf("postgres://%s:%s@%s", user, password, url)
-
-	return url, user, password, nil
+	connURL = interpolateConnectionURL(url, user, password)
+	return connURL, url, user, password, nil
 }
 
-func TestPostgreSQL_Initialize(t *testing.T) {
+func TestRedshift_Initialize(t *testing.T) {
 	if os.Getenv(vaultACC) != "1" {
 		t.SkipNow()
 	}
 
-	url, _, _, err := redshiftEnv()
+	connURL, _, _, _, err := redshiftEnv()
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	connectionDetails := map[string]interface{}{
-		"connection_url":       url,
-		"max_open_connections": 5,
+		"connection_url":       connURL,
+		"max_open_connections": 73,
 	}
 
-	db := newRedshift(true)
-	_, err = db.Init(context.Background(), connectionDetails, true)
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	db := newRedshift()
+	resp := dbtesting.AssertInitialize(t, db, dbplugin.InitializeRequest{
+		Config:           connectionDetails,
+		VerifyConnection: true,
+	})
 
 	if !db.Initialized {
 		t.Fatal("Database should be initialized")
 	}
-
-	err = db.Close()
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	if resp.Config["max_open_connections"] != 73 {
```

Comment:
Instead of checking a single field in the returned config, I recommend making an `expectedConfig` object that is a copy/duplicate of the `connectionDetails` map above (a copy/duplicate since the `Initialize` call could manipulate the map without us realizing it) and then doing a `if !reflect.DeepEquals` on it